strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/providers/io/AbstractRepositoryProvider.java

```diff
@@ -144,7 +144,7 @@ public void onBeforeWrite(RepositoryStreamWriteContext ctx) throws IOException
             return;
         }
 
-        Repository repository = repositoryPath.getRepository();
+        RepositoryData repository = repositoryPath.getRepository();
```

Comment:
This sort of breaks the naming convention, if you see what I mean, but in a way I get it...

Reply:
@carlspring please have a look below comment